core/src/main/java/org/everit/json/schema/loader/SchemaLoader.java

```diff
@@ -122,11 +122,23 @@ public SchemaLoaderBuilder draftV7Support() {
 
         private void setSpecVersion(SpecificationVersion specVersion) {
             this.specVersion = specVersion;
-            specVersion.defaultFormatValidators().forEach(this::addFormatValidator);
-            //            this.formatValidators = new HashMap<>(specVersion.defaultFormatValidators());
+        }
+
+        private Optional<SpecificationVersion> specVersionInMetaSchema() {
+            Optional<SpecificationVersion> specVersion = Optional.empty();
+            if (schemaJson instanceof Map) {
+                Map<String, Object> schemaObj = (Map<String, Object>) schemaJson;
+                Object schemaValue = schemaObj.get("$schema");
+                if (schemaValue != null) {
+                    specVersion = Optional.of(SpecificationVersion.getByMetaSchemaUrl((String) schemaValue));
+                }
```

Comment:
The above 4 lines could be simplified to `specVersion = Optional.ofNullable(schemaObj.getString("$schema")).map((SpecificationVersion::getByMetaSchemaUrl))`

Reply:
Small modification: schemaObj.get("$schema") instead of getString and a cast to String.